test/contracts/kill_switch/kill_switch.test.js

```diff
@@ -0,0 +1,963 @@
+  let kernelBase, aclBase, appBase, killSwitchBase, issuesRegistryBase, daoFactory
+  let kernelWithoutKillSwitchBase, kernelWithNonCompliantKillSwitchBase, failingKillSwitchBase
+  let CORE_NAMESPACE, KERNEL_APP_ID, APP_MANAGER_ROLE, SET_SEVERITY_ROLE, SET_DEFAULT_ISSUES_REGISTRY_ROLE, SET_ISSUES_REGISTRY_ROLE, SET_ALLOWED_INSTANCES_ROLE, SET_DENIED_BASE_IMPLS_ROLE, SET_HIGHEST_ALLOWED_SEVERITY_ROLE
+
+  before('deploy base implementations', async () => {
+    // real
+    kernelBase = await Kernel.new(true) // petrify immediately
+    aclBase = await ACL.new()
+    registryFactory = await EVMScriptRegistryFactory.new()
+    killSwitchBase = await KillSwitch.new()
+    issuesRegistryBase = await IssuesRegistry.new()
+
+    // mocks
+    appBase = await KillSwitchedApp.new()
+    failingKillSwitchBase = await FailingKillSwitchMock.new()
+    kernelWithoutKillSwitchBase = await KernelWithoutKillSwitchMock.new()
+    kernelWithNonCompliantKillSwitchBase = await KernelWithNonCompliantKillSwitchMock.new()
+  })
+
+  before('load constants and roles', async () => {
+    CORE_NAMESPACE = await kernelBase.CORE_NAMESPACE()
+    KERNEL_APP_ID = await kernelBase.KERNEL_APP_ID()
+    APP_MANAGER_ROLE = await kernelBase.APP_MANAGER_ROLE()
+    SET_SEVERITY_ROLE = await issuesRegistryBase.SET_SEVERITY_ROLE()
+    SET_DEFAULT_ISSUES_REGISTRY_ROLE = await killSwitchBase.SET_DEFAULT_ISSUES_REGISTRY_ROLE()
+    SET_ISSUES_REGISTRY_ROLE = await killSwitchBase.SET_ISSUES_REGISTRY_ROLE()
+    SET_ALLOWED_INSTANCES_ROLE = await killSwitchBase.SET_ALLOWED_INSTANCES_ROLE()
+    SET_DENIED_BASE_IMPLS_ROLE = await killSwitchBase.SET_DENIED_BASE_IMPLS_ROLE()
+    SET_HIGHEST_ALLOWED_SEVERITY_ROLE = await killSwitchBase.SET_HIGHEST_ALLOWED_SEVERITY_ROLE()
+  })
+
+  context('when the kernel version does not support kill-switch logic', async () => {
+    before('create DAO factory', async () => {
+      daoFactory = await DAOFactory.new(kernelWithoutKillSwitchBase.address, aclBase.address, ZERO_ADDRESS, registryFactory.address)
+    })
+
+    beforeEach('deploy DAO without a kill switch and create kill-switched sample app', async () => {
+      const daoFactoryReceipt = await daoFactory.newDAO(root)
+      dao = Kernel.at(getEventArgument(daoFactoryReceipt, 'DeployDAO', 'dao'))
+      acl = ACL.at(await dao.acl())
+      await acl.createPermission(root, dao.address, APP_MANAGER_ROLE, root, { from: root })
+
+      const appReceipt = await dao.newAppInstance('0x1236', appBase.address, '0x', false, { from: root })
+      app = KillSwitchedApp.at(getNewProxyAddress(appReceipt))
+      await app.initialize(owner)
+    })
+
+    it('executes the call', async () => {
+      await app.write(10, { from: owner })
+      assert.equal(await app.read(), 10)
+    })
+  })
+
+  context('when the kernel version does support kill-switch logic', async () => {
+    const SAMPLE_APP_ID = '0x1236000000000000000000000000000000000000000000000000000000000000'
```

Comment:
Perhaps this constant could be lifted up, I notice we use a similar one in https://github.com/aragon/aragonOS/pull/518/files#diff-20fa58b7946252b898fd2dc455170cf6R65.